### PR TITLE
integration: Update CRI integration tests for new containerd config

### PR DESCRIPTION
**Pull request by Bevisy**
Update CRI integration tests for new containerd config

Fixes: #3682

Refer to [cri/config.md](https://github.com/containerd/containerd/blob/main/docs/cri/config.md)

Signed-off-by: Binbin Zhang <binbin36520@gmail.com>

### CODE REVIEW

**Comment by Bevisy** [comment]
I'm glad to fix this issue, please approve to run ci workflows. 
PTAL @jodh-intel. Thanks.

**Comment by Bevisy** [comment]
The reason for the ci failure is that tabs and spaces are mixed, fixed now

**Comment by GabyCT** [comment]
@Bevisy could you please rebase your PR to fix the conflicting issues? thanks

**Comment by Bevisy** [comment]
@GabyCT Thanks for reminding me, I have fixed the conflict and force pushed.

File: integration/containerd/cri/integration-tests.sh [hunk 1]
```diff
@@ -127,19 +127,20 @@ create_containerd_config() {
...
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.${runtime}]
+          runtime_type = "${runtime_type}"
+          $( [ $kata_annotations -eq 1 ] && \
+          echo 'pod_annotations = ["io.katacontainers.*"]' && \
+          echo -n '          container_annotations = ["io.katacontainers.*"]'
+          )
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.${runtime}.options]
+            BinaryName = "${containerd_runtime}"
+[plugins."io.containerd.runtime.v1.linux"]
```
Comment (liubin) [inline integration/containerd/cri/integration-tests.sh:142]: Should add some spaces to align to `[plugins."io.containerd.grpc.v1.cri"]`?

Comment (Bevisy) [reply integration/containerd/cri/integration-tests.sh]: Yes, it should, thanks :smile:

**Comment by Bevisy** [comment]
Maybe I need more time to figure out why ci is failing 😞 
```sh
23:49:35 [integration-tests.sh:203] INFO: restart containerd service
23:49:35 [integration-tests.sh:288] ERROR: The VM memory size 16819544064 before update is not right
23:49:35 Makefile:138: recipe for target 'cri-containerd' failed
23:49:35 make: *** [cri-containerd] Error 1
23:49:35 [run.sh:40] ERROR: sudo -E PATH=/tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-initrd-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games bash -c make cri-containerd
23:49:35 [jenkins_job_build.sh:193] ERROR: .ci/run.sh
23:49:35 Build step 'Execute shell' marked build as failure
```

**Review by jodh-intel — APPROVED** [review]
Thanks @Bevisy.

lgtm

**Comment by Bevisy** [comment]
> Thanks @Bevisy.
> 
> lgtm

I don't understand why CI fails on ubuntu and debian, but works fine on centos. I tried to dig for the reason, but I have to admit that I didn't find it.
```sh
15:40:25 [integration-tests.sh:218] INFO: restart containerd service
15:40:25 [integration-tests.sh:303] ERROR: The VM memory size 268508655616 before update is not right
15:40:26 make: *** [Makefile:138: cri-containerd] Error 1
15:40:26 [run.sh:44] ERROR: sudo -E PATH=/home/jenkins/.cargo/bin:/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-ARM-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games bash -c make cri-containerd
15:40:26 [jenkins_job_build.sh:215] ERROR: .ci/run.sh
15:40:26 Build step 'Execute shell' marked build as failure
```

File: integration/containerd/cri/integration-tests.sh [hunk 2]
```diff
@@ -122,25 +124,45 @@ create_containerd_config() {
 	if [ "${runtime_type}" == "${default_runtime_type}" ];then
 		local containerd_runtime=$(command -v "${runtime}")
 	fi
+	
+	if [ "${runtime_type}" == "${kata_runtime_type}" ];then
+		local containerd_runtime=${kata_runtime_bin}
+	fi
```
Comment (Jakob-Naucke) [inline integration/containerd/cri/integration-tests.sh:130]: Are these changes necessary up to here? AFAIU the `SHIMV2_TEST` variable already handles all of this.

File: integration/containerd/cri/integration-tests.sh [hunk 3]
```diff
@@ -122,25 +124,45 @@ create_containerd_config() {
...
+
+if [ $kata_annotations -eq 1 ]; then
+	cat << EOT | sudo tee "${CONTAINERD_CONFIG_FILE}"
 [plugins]
-  [plugins.cri]
-    [plugins.cri.containerd]
-        default_runtime_name = "$runtime"
-      [plugins.cri.containerd.runtimes.${runtime}]
-        runtime_type = "${runtime_type}"
-        $( [ $kata_annotations -eq 1 ] && \
```
Comment (Jakob-Naucke) [inline integration/containerd/cri/integration-tests.sh:135]: I preferred this logic TBH, way fewer LOC...

**Review by Jakob-Naucke — COMMENTED** [review]
Thank you @Bevisy -- I left some comments. The CI failures are definitely surprising, let me TAL.

**Comment by Jakob-Naucke** [comment]
@Bevisy I run into another failure with the virtio-mem tests atm -.- but were you able to verify these tests on your machine? Most of the passing tests actually don't run these at all (because they're on CRI-O or they test something more specific), s390x might seem like an exception, except it doesn't test virtio-mem either.

File: integration/containerd/cri/integration-tests.sh (hunk 2 above)
Comment (Bevisy) [reply integration/containerd/cri/integration-tests.sh]: Ok, I have already delete this part.

**Comment by jodh-intel** [comment]
@Bevisy - please could you update this PR as it is now conflicted.

**Comment by jodh-intel** [comment]
Ping @Bevisy!

**Comment by jodh-intel** [comment]
Hi @Bevisy - please can you rebase to resolve the conflict on this PR?

**Comment by Bevisy** [comment]
> Hi @Bevisy - please can you rebase to resolve the conflict on this PR?

hanks for the reminder, will deal with it later.

**Comment by Bevisy** [comment]
I'm sorry, but I'm finding it difficult to find the cause of the CI failures here, and I'm going to turn this modification off. I'll consider turning it back on when I have enough time to do so. At the moment the main work time is spent on GPU support.
```sh
12:56:26 [integration-tests.sh:296] ERROR: The VM memory size 16788598784 before update is not right
```
